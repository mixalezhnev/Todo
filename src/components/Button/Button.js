import React, { Component } from 'react';
import '../../style/Button.css';

class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    alert('Oh yea ;)');
  }

  render(props) {
    return (
      <a href="#" className="Button" onClick={this.handleClick}> 
        { this.props.text }
      </a>
    )
  }
}

export default Button;