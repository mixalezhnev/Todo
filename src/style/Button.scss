.Button {
  width: 100px;
  min-height: 36px;
  border: 0;
  outline: none;
  text-align: center;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 0 11px;
  background-color: #1B89DE;
  border-radius: 4px;
  font-family: sans-serif;
  font-weight: 600;
  text-decoration: none;
  color: white;
  margin-bottom: 5px;
}

.Button-red {
  background: #FB8888;
}

.Button-grey {
  background: #5E6472;
}