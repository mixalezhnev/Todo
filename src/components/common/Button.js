import React from 'react';

import './Button.res/style.css'

const Button = (props) => {
  const {type, styles} = props;
  
  return  <button style={styles} className={`Button Button-${type}`}></button>;
}

Button.defaultProps = {
  type: 'add',
  styles: { 
    top: 0,
    left: 0
  }
}

export default Button;
