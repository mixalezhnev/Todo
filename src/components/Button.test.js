import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

const setup = () => {
  const props = {
    onClick: () => {},
    style: {
      margin: '0 5px'
    },
    type: 'swap'
  };

  const enzymeWrapper = shallow(<Button {...props}/>)

  return {
    props,
    enzymeWrapper
  }
};

describe('components', () => {
  describe('Button', () => {
    it('should render self', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});