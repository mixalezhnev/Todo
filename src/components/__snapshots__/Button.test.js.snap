exports[`components Button should render self 1`] = `
<button
  className="my-button-component button-swap"
  onClick={[Function]}
  style={
    Object {
      "margin": "0 5px",
    }
  } />
`;
